### DISCUSSION TITLE
Getting and Setting custom field values

### DISCUSSION THREAD

**Discussion by joegiglio** [pycontribs/jira#1433] [General]
Very nice package.  Works great for my needs, EXCEPT I can't figure out a simple syntax for getting and setting values for custom fields. 

The suggestion [[here](https://community.atlassian.com/t5/Jira-questions/how-to-get-customfield-value-in-python-jira-api/qaq-p/1166668)] seems to work but it seems overly complex.  Is there no syntax for .fields.custom_field_id or similar?

**Comment by adehad** [comment]
Unfortunately at the moment there isn't anything much better.

We do have a recently added [`get_field()`](https://github.com/pycontribs/jira/pull/1102) function, not sure if that helps you:
```python
issue = self.jira.issue("my_issue-number")
issue.get_field("customfield_1234")
```

Happy to take suggestions and PRs for improvements !

**Reply by joegiglio** [reply]
Your suggestion worked fine. You may want to update the docs to clarify this point for getting values for custom fields.

Any tips for SETTING a custom field value?

**Reply by adehad** [reply]
Unfortunately nothing for setting fields yet. If you do end up implementing something you are happy to share would be great to add to the package

**Reply by adehad** [reply]
Technically the `update()` method may be an option, but the problem with custom fields is that the data type also needs to be queried (for example adding to a label like field would use `issue.add_field_value()`) so fields like date's and times would need the user to enter the value correctly.

Here is a rough example again:
```python
issue.update(fields={"update": {"customfield_1234": "new_value_that_must_be_valid_for_this_field_type"}})
```
